Doc/library/imath.rst

```diff
@@ -0,0 +1,99 @@
+:mod:`imath` --- Mathematical functions for integer numbers
+===========================================================
+
+.. module:: imath
+   :synopsis: Mathematical functions for integer numbers.
+
+.. versionadded:: next
+
+--------------
+
+This module provides access to the mathematical functions for integer arguments.
+These functions accept integers and objects that implement the
+:meth:`~object.__index__` method which is used to convert the object to an integer
```

Comment:
In the same way we could simplify now description of argument processing in the math module.  Except for a special snowflake "log" - all use PyFloat_AsDouble().  (Well, few functions also try special dunder methods.)

Though, probably it's for following prs.